sovtokenfees/sovtokenfees/req_handlers/batch_handlers/fee_batch_handler.py

```diff
@@ -0,0 +1,81 @@
+from sovtoken import TOKEN_LEDGER_ID
+from sovtoken.constants import UTXO_CACHE_LABEL
+from sovtoken.request_handlers.token_utils import commit_to_utxo_cache
+from sovtokenfees.constants import FEES
+from sovtokenfees.req_handlers.fees_utils import BatchController
+
+from common.serializers.serialization import txn_root_serializer
+from plenum.common.constants import DOMAIN_LEDGER_ID
+from plenum.common.txn_util import get_seq_no, get_type
+from plenum.server.batch_handlers.batch_request_handler import BatchRequestHandler
+from plenum.server.batch_handlers.three_pc_batch import ThreePcBatch
+from stp_core.common.log import getlogger
+
+logger = getlogger()
+
+
+class FeeBatchHandler(BatchRequestHandler):
+    def __init__(self, database_manager, batch_controller: BatchController, token_tracker):
+        super().__init__(database_manager, DOMAIN_LEDGER_ID)
+        self._batch_controller = batch_controller
+        self._token_tracker = token_tracker
+
+    @property
+    def token_state(self):
+        return self.database_manager.get_state(TOKEN_LEDGER_ID)
+
+    @property
+    def token_ledger(self):
+        return self.database_manager.get_ledger(TOKEN_LEDGER_ID)
+
+    @property
+    def utxo_cache(self):
+        return self.database_manager.get_store(UTXO_CACHE_LABEL)
+
+    def post_batch_applied(self, three_pc_batch, prev_handler_result=None):
+        self._token_tracker.apply_batch(self.token_state.headHash,
+                                        self.token_ledger.uncommitted_root_hash,
+                                        self.token_ledger.uncommitted_size)
+        if self._batch_controller.fees_in_current_batch > 0:
+            state_root = self.token_state.headHash
+            self.utxo_cache.create_batch_from_current(state_root)
```

Comment:
Should we move utxo_cache-related logic into a separate BatchHandler, such as `UtxoCacheFeeBatchHandler`?

Reply:
I think we should separate logic, but we can use existing UTXOBatchHandler from sovtoken

Reply:
yes

Reply:
too tied for now